src/object-store/src/object_store_manager.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use snafu::ensure;
+
+use crate::error::Result;
+use crate::{error, ObjectStore};
+
+#[derive(Clone)]
+pub struct ObjectStoreManager {
+    stores: HashMap<String, ObjectStore>,
+    global_object_store: String,
+}
+
+impl ObjectStoreManager {
+    pub fn try_new(
+        stores: HashMap<String, ObjectStore>,
+        global_object_store: &str,
+    ) -> Result<Self> {
+        ensure!(
+            stores.keys().any(|key| key == global_object_store),
+            error::GlobalStorageNotFoundSnafu {
+                global_object_store: global_object_store.to_string()
+            }
+        );
```

Comment:
suggestion: The global (default) object store is commonly used, we can keep the struct in `global_object_store`. We always provide the global one in new. Other object stores are some "extensions" so we can register them later.

e.g.
```rust
struct ObjectStoreManager {
    stores: HashMap<String, ObjectStore>,
    default_object_store: ObjectStore,
}

impl ObjectStoreManager {
    /// Creates a new manager with the default object store.
    pub fn new(name: &str, default_object_store: ObjectStore) -> Self {
        // Add the default object store to the map.
        let stores = ...;
        Self { stores, default_object_store, }
    }

    /// Adds an object store to the manager.
    pub fn add_object_store(&mut self, name: &str, object_store: ObjectStore) {}
```

Reply:
I'd like to ensure that `ObjectStoreManager` has all the object stores at the `new` since the storage options is loaded at startup.
I guess that we'd like to avoid to dynamically change the settings at runtime.

What do you think about this?

Reply:
> I guess that we'd like to avoid to dynamically change the settings at runtime.

In the example above, adding an object store requires `&mut self` so `Arc<ObjectStoreManager>` doesn't allow adding new object store. I separate the `try_new` to avoid handling error while the default key doesn't exist in the object store map.

But providing all object stores in `try_new` is reasonable.